st3/lsp_utils/node_runtime.py

```diff
@@ -97,13 +97,13 @@ def _resolve_node_runtime(
                                 NO_NODE_FOUND_MESSAGE.format(package_name=package_name), 'Download Node.js'):
                             log_lines.append(' * Download skipped')
                             continue
+                    # Remove outdated runtimes.
+                    if path.isdir(runtime_dir):
+                        for directory in next(os.walk(runtime_dir))[1]:
+                            old_dir = path.join(runtime_dir, directory)
+                            print('[lsp_utils] Deleting outdated Node.js runtime directory "{}"'.format(old_dir))
+                            shutil.rmtree(old_dir, ignore_errors=True)
```

Comment:
Even if it shouldn't be in the original try/except block, it should probably still be in some exception block?

In other cases we `continue` on exception so it might make sense to do here also? Maybe... saying this without recalling how things works here exactly.

Also, with or without try/except, what state are we gonna end up with when this only partially deletes the runtime dir? Would this still result in things crashing on next start without auto-healing itself?

Reply:
After some further investigation, I found that the root cause is not due to locked file. It's because the "long path (>255 chars)" issue on Windows.

In the original error msg: ` * Failed downloading: [WinError 3] File not found: 'C:\\Users\\jfcherng\\AppData\\Local\\Sublime Text\\Package Storage\\lsp_utils\\node-runtime\\18.17.1\\cache\\yarn\\v6\\npm-copilot-node-server-1.11.0-7eed36a428960e570c06cff8ccd86d70450cacdf-integrity\\node_modules\\copilot-node-server\\copilot\\dist\\compiled\\darwin\\arm64\\kerberos.node'`

The path is 268-char long. The solution is convert the path into its UNC form as stated in code comments.

Reply:
Can you also pick the 60dcba28532009334b1b0ec8ea7b5f90927bc0d3 changes from this repo? I can't push to your organization

Reply:
Done.